### PR TITLE
rc parser improvements

### PR DESCRIPTION
**Pull request by realtaraharris**
This PR adds support for comments and whitespace in the rc parser (used to parse `.npmrc` files). It also makes some minor changes to appease our linter. Finally, it adds a script (`detailed-coverage`), which provides a detailed coverage report to aid development.

### CODE REVIEW

File: commands/signin.js [hunk 1]
```diff
@@ -41,11 +41,11 @@ function exchangeAuthCodeForAccessToken (authorizationCode) {
     url: exchangeUri,
     headers: { 'content-type': 'application/json' },
     body: JSON.stringify({
-      "grant_type": "authorization_code",
-      "client_id": clientId,
-      "code_verifier": verifier,
-      "code": authorizationCode,
-      "redirect_uri": redirectUri
+      'grant_type': 'authorization_code',
```
Comment (pmuellr) [inline commands/signin.js]: technical nit: don't need quotes for keys unless they aren't valid identifiers

Comment (realtaraharris) [reply commands/signin.js]: sure, I'm aware... the way this pattern plays out is that some JSON gets pasted in there, then variables threaded in, then someone notices that it doesn't lint with the double-quotes...

I'll change it to match your suggestion here

File: commands/signin.js [hunk 2]
```diff
@@ -58,6 +58,7 @@ function stripProtocol (certifiedModulesUrl) {
 }
 
 function accessTokenReceived (error, response, info) {
+  if (error) { console.error(error) }
```
Comment (pmuellr) [inline commands/signin.js]: this doesn't seem right.  Just prints the error (with no context on what it's from), and continues with further processing.

File: lib/rc.js [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 
 function stringify (parsedData, commentChar) {
-  return Object.keys(parsedData).map(key => {
-    const v = parsedData[key]
-    const finalKey = v.comment ? `${commentChar}${key}` : key
-    return `${finalKey}=${v.value}`
-  }).join('\n') + '\n'
+  return parsedData.map(line => {
+    const finalKey = line.comment ? `${commentChar}${line.key}` : line.key
+    const separator = finalKey && line.value ? '=' : ''
```
Comment (pmuellr) [inline lib/rc.js:7]: this doesn't see quite right - if a line is a comment, why muck with it, and assume that it has key / value parts?  In any case,  looks like a commented out setting would end up having the `=` removed from it.

Comment (realtaraharris) [reply lib/rc.js:7]: Every line gets turned into an array of objects, with a definite structure:

```
    if (parts.length === 2) {
      const { key, comment } = parseKey(parts[0].trim())
      const value = parts[1].trim()
      if (key) {
        result.push({ key, value, comment })
      }
    } else if (parts.length === 1) {
      const { key, comment } = parseKey(parts[0].trim())
      result.push({ key, value: '', comment })
    }
```

So this really isn't an issue. Also note that the test case at the end of `rc-test.js` covers this exact thing - it checks to see if the fixture (`tests/fixtures/pmuellr`) comes back intact after going through a serialize/deserialize cycle, and indeed it does.

**Review by julianduque — APPROVED** [review]
LGTM

**Pull request merged**